### ISSUE TITLE
RemoveUnnecessaryImports diagnostics should not be reported when we do not have all information

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#78737]
Repro: paste the following into VS Code

```cs
using Newtonsoft.Json;

class C
{
    public void M()
    {
        var jsonString = """{"name":"Rikki"}""";
        var jsonObject = JsonConvert.DeserializeAnonymousType(jsonString, new { name = "" })!;
        Console.Write(jsonObject.name);
        Console.Write(BAD);
    }
}
```

Expected: No "unnecessary import" diagnostic is reported on `using Newtonsoft.Json`.
Actual: The diagnostic is reported (and the using is greyed out).

Rationale: For a non-file-based-program, we set HasAllInformation = false, indicating semantic diagnostics should not be reported. This is respected already for compiler semantic diagnostics: no diagnostic is being reported for `BAD`. This should apply to the unnecessary imports analyzer also, for example. We don't have all information, so we don't know whether this file is making use of this `using` directive or not.

**Comment by CyrusNajmabadi** [comment]
I think there's a difference here. Compiler diagnostics show as errors to the user, indicating something is wrong. Our own analyzer diagnostics drive features that users still want (like converting from block to expression bodies). These are not shown as errors to the user.

**Comment by CyrusNajmabadi** [comment]
> Converting from block to expression body in particular sounds like a syntax diagnostic. 

Currently it isn't.  The only think that are truly syntax are the full 'syntax tree analyzers' which also pass a flag stating they don't need the semantic model.  99% of our analyzers are not this form (basically, only formatting, add accessibility modifiers, file header analyzers).

**Comment by RikkiGibson** [comment]
What should we do for the RemoveUnusedImports case then? I think as-is, we are simply missing references and source files, and the analyzer doesn't know that is the case. Is there some mechanism we should be using to propagate in `HasAllInformation` and check it?

**Comment by RikkiGibson** [comment]
The current state of things is that when user opens a random .cs file in VS Code, the editor greys out and effectively suggests to remove a bunch of using directives in it, when there isn't a strong basis for making that suggestion. This happens for non-file-based-program files, as well as for file-based-programs while in the "primordial" state (the several seconds while waiting for a design-time build to finish), as well as for files which are actually part of an ordinary project, but the project hasn't been loaded yet. In my opinion, that is a bad user experience.

I think this issue shouldn't generalize to all analyzers, I will edit the title to reflect the specific situation.

**Comment by RikkiGibson** [comment]
> I'm curious why it takes 10 seconds for the project to load. That seems... crazy bad?

That was definitely an issue with my setup. However, I am concerned there are other conditions I don't understand yet, which cause us to linger in this state.

**Comment by RikkiGibson** [comment]
Closing out as a duplicate of #79005 which seems to be showing a more pressing customer scenario.
